projectq/ops/_gates.py

```diff
@@ -117,6 +117,22 @@ def __str__(self):
 Tdag = Tdagger = get_inverse(T)
 
 
+class SqXGate(BasicGate):
+    """ Square-root X gate class """
+    @property
+    def matrix(self):
+        return 0.5 * np.matrix([[1+1j, 1-1j], [1-1j, 1+1j]])
+
+    def tex_str(self):
+        return r'$\sqrt{\neg}$'
+
+    def __str__(self):
+        return "SqX"
+
+SqX = SqXGate()
+SqXdag = SqXdagger = get_inverse(SqX)
```

Comment:
I haven't seen many uses of this gate which would justify an own gate name.
get_inverse or with Dagger(eng) might be easy enough in case it is ever needed

Reply:
Let's discuss more on this topic. In the future, we may need a standard reference to unify the table of gates and the naming system. 

What about this book
http://www.springer.com/cda/content/document/cda_downloaddocument/9781846288869-c2.pdf?SGWID=0-0-45-1039041-p173733620

Or, have you already got one?

Reply:
We tend to follow gate definition of Nielsen and Chuang as they are widely accepted in literature for the standard gates. For more exotic gates, there might be discussions.

In any case because we define the gate matrix, everyone can and should check that they use the same convention. E.g. some decompositions of ProjectQ rely on a paper by Barenco et al. in which different definitions for the basic gates have been used than in Nielsen and Chuang or in  ProjectQ. So it required some adaptations...

Daggered gates are only named if they are frequently used such as T and Tdagger but that is the only one which comes to my mind. For the moment I would suggest to not name daggered gates but let users write them by using `get_inverse()` or `with Dagger`. They can still name the gate in the user code if they like... And should a particular inverse gate be used a lot in the community, we can add the name very quickly.

Reply:
As your suggested, I will remove sqXdag, and rename SqX to SqrtX.

btw, one of my reasons to add these two square root gates is they have their own widely accepted symbols. So the automatically generated Tikz graph can look better. This is an aesthetic criteria :)

Reply:
👍  Thanks.
I agree, SqrtX and SqrtSwap are widely used and I would have had to implement them soon myself. So I am very happy that you did this work already  (including the nice Tikz graph symbols 💯 )